### PR TITLE
BUGFIX: Allow to configure Imagick resource limits

### PR DESCRIPTION
**Pull request by kitsunet**
In order to be able to run in confined server environments a resource
limit for Imagick can be configured now as otherwise Imagick is very
generous in it's resource usage.

### CODE REVIEW

**Comment by kitsunet** [comment]
Seems for styleCI I probably need to change this package to psr-2 which apparently was never done...

**Comment by kdambekalns** [comment]
Ok, StyleCI still fails, but at least it "fails correctly" now :)

**Review by kdambekalns — APPROVED** [review]
Looks good… I guess @robertlemke can actually test it?!

File: Classes/TYPO3/Imagine/ImagineFactory.php [hunk 1]
```diff
@@ -20,32 +20,34 @@
  */
 class ImagineFactory extends AbstractImagineFactory
 {
-    /**
+	/**
      * @Flow\Inject
```
Comment (aertmann) [inline Classes/TYPO3/Imagine/ImagineFactory.php]: this one should get fixed too right? not sure what's going on with the indentation, seems mixed up?

File: Configuration/Settings.yaml [hunk 2]
```diff
@@ -4,4 +4,10 @@ TYPO3:
     profile:
       'RGB': 'color.org/sRGB_IEC61966-2-1_black_scaled.icc'
       'CMYK': 'Adobe/CMYK/USWebUncoated.icc'
-      'Grayscale': 'colormanagement.org/ISOcoated_v2_grey1c_bas.ICC'
\ No newline at end of file
+      'Grayscale': 'colormanagement.org/ISOcoated_v2_grey1c_bas.ICC'
+    driverSpecific:
+      Imagick:
+        # define a map of limits, where the key is an int (one of the "IMagick::RESOURCETYPE_*" constants)
```
Comment (aertmann) [inline Configuration/Settings.yaml]: `Imagick::` no?

File: Classes/TYPO3/Imagine/ImagineFactory.php [hunk 3]
```diff
@@ -67,11 +69,38 @@ public function create($className = 'Imagine')
...
+	 * @return void
+	 */
+	protected function configureImagickSettings() {
+		if (!isset($this->settings['driverSpecific']['Imagick'])) {
+			return;
+		}
 
-        return $object;
-    }
+		$limits = $this->settings['driverSpecific']['Imagick']['limits'] ?? [];
```
Comment (aertmann) [inline Classes/TYPO3/Imagine/ImagineFactory.php]: null coalesce operator only available in PHP 7.0+, Imagine 2.0 requires Flow ~3.0 which is 5.5

**Review by aertmann — CHANGES_REQUESTED** [review]

**Review by aertmann — APPROVED** [review]
LGTM by reading

**Review by kdambekalns — APPROVED** [review]

**Pull request merged**